### PR TITLE
feat: `PostResponse` supports charset other than `UTF-8`

### PR DESCRIPTION
**Pull request by joyyir**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#238 
___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.
`PostResponse` by default converts the body to bytes in `UTF-8`.
It would be nice to support other charsets.

- Source branch
main

- Related commits and pull requests

- Target branch
main

### CODE REVIEW

**Review by nic-chen — COMMENTED** [review]
Thank you very much for the contribution, please add some test cases for the change.

**Comment by joyyir** [comment]
Added test case. Thank you.

File: runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/PostResponse.java [hunk 1]
```diff
@@ -53,7 +57,7 @@ public ByteBuffer encode() {
 
         int bodyIndex = -1;
         if (StringUtils.hasText(body)) {
-            byte[] bodyBytes = body.getBytes(StandardCharsets.UTF_8);
+            byte[] bodyBytes = body.getBytes(charset);
```
Comment (nic-chen) [inline runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/PostResponse.java:60]:
```suggestion
            byte[] bodyBytes = body.getBytes(this.charset);
```
should be better?

**Review by nic-chen — DISMISSED** [review]
LGTM

**Review by Gallardot — DISMISSED** [review]

File: runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/PostResponse.java (hunk 1 above)
Comment (joyyir) [reply runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/PostResponse.java]: I've reflected your review. Thank you.

**Review by nic-chen — APPROVED** [review]

**Comment by nic-chen** [comment]
@joyyir thanks!

**Pull request merged**